src/go/tokengenerator/token_generator.go

```diff
@@ -12,26 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package util
+package tokengenerator
 
 import (
+	"fmt"
 	"io/ioutil"
+	"net/http"
 	"sync"
 	"time"
 
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
+	"github.com/gorilla/mux"
 	"golang.org/x/oauth2"
 	"golang.org/x/oauth2/google"
 )
 
 var (
 	_GOOGLE_API_SCOPE = []string{
+		// Call servicemanagement to fetch service config.
 		"https://www.googleapis.com/auth/service.management.readonly",
+		// Call servicecontrol to get latest rollout id.
+		"https://www.googleapis.com/auth/servicecontrol",
```

Comment:
e need to add QuotaController

Reply:
"https://www.googleapis.com/auth/servicecontrol" look enough. 
https://developers.google.com/identity/protocols/oauth2/scopes#servicecontrol

Adding e2e test for quota.

Reply:
Thanks, we need to make sure quotaController works for service account key file

Reply:
Quota test has been checked in.